src/converter/one/proto/Proto.java

```diff
@@ -71,7 +71,7 @@ public void writeInt(int n) {
         int length = n == 0 ? 1 : (38 - Integer.numberOfLeadingZeros(n)) / 7;
         ensureCapacity(length);
 
-        while (n > 0x7f) {
+        while (n & 0x7f != n) {
```

Comment:
Why can a number be negative? In what cases?

Reply:
The case I hit was `nanosSinceLastSample` in `jfr2pprof.java` being negative.

I don't know if that is supposed to happen; all I can tell you is that it happened, with async-profiler 3.0 and the default clock source.